rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelScheduler.java

```diff
@@ -0,0 +1,217 @@
+package com.lithium.dbi.rdbi.recipes.scheduler;
+
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import redis.clients.jedis.Tuple;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.LongSupplier;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+/**
+ * Similar to {@link StateDedupedJobScheduler}, except that it includes a separate "channel"
+ * dimension and attempts to maintain fairness among jobs in a particular by round-robining
+ * through all channels.
+ *
+ * Includes support for methods from {@link StateDedupedJobScheduler} and it's abstract hierarchy,
+ * however does not extend from that tree because some methods require the channel parameter
+ *
+ * it may prove useful to create a MultiChannelScheduler.ForChannel that extends from the {@link AbstractDedupJobScheduler} hierarchy
+ *
+ * // TODO This doesn't yet incorporate concepts from https://github.com/lithiumtech/rdbi/commit/6bbf2eeb49b87b71655f24fa9b797300b37b6797, that will be tackled separately
+ */
```

Comment:
Suggest adding example usage for scheduling and then reserving jobs.  For example, after I've reserved a job and finished it, do I need to ack completion back to this api?  Also, if I reserve a job and fail to process it b/c my jvm is restarted, what recourse do I have in the api  to let some other jvm pick up the scheduled job?

Reply:
sure, i can expand on it some.. there's a fair bit of implicit knowledge required based on how we use the other schedulers. 

as for handling the failure scenarios you described there's nothing at this level that dictates you it's done. The only recourse available is to look for expired running jobs, and then decide if they should be retried or not.